File: BOJ/2823.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;
char board[11][11];
int r, c;
bool ok;
int gox[4] = {-1,0,1,0};
int goy[4] = {0,1,0,-1};
int main(){
	scanf("%d%d", &r, &c);
	for (int i = 1; i <= r; i++) {
		for (int j = 1; j <= c; j++) {
			scanf(" %c", &board[i][j]);
		}
	}
	for (int i = 1; i <= r; i++) {
		for (int j = 1; j <= c; j++){
			if (board[i][j] == '.'){
				int cnt = 0;
				for (int k = 0; k < 4; k++) {
					int nx = i + gox[k];
					int ny = j + goy[k];
					if (nx<1 || nx>r || ny<1 || ny>c || board[nx][ny] == 'X') continue;
					if (board[nx][ny] == '.') cnt++;
				}
				if (cnt <= 1) {
					ok = true;
					break;
				}
			}
		}
		if (ok) break;
	}
	if (ok) printf("1\n");
	else printf("0\n");
	return 0;
}

File: BOJ(Java 공부용)/2823.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 0, 1, -1};
    static int[] goy = {1, -1, 0, 0};
    static int counter = 1;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken()), m = Integer.parseInt(st.nextToken());
        char[][] board = new char[n][m];
        for (int i = 0; i < n; i++) {
            String s = br.readLine();
            for (int j = 0; j < s.length(); j++) {
                if (s.charAt(j) == ' ') break;
                board[i][j] = s.charAt(j);
            }
        }

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (board[i][j] == '.') {
                    int cnt = 0;
                    for (int k = 0; k < 4; k++) {
                        int nx = i + gox[k], ny = j + goy[k];
                        if (nx < 0 || nx >= n || ny < 0 || ny >= m || board[nx][ny] == 'X') continue;
                        cnt++;
                    }
                    if (cnt <= 1) {
                        bw.write("1");
                        bw.flush();
                        bw.close();
                        return;
                    }
                }
            }
        }
        bw.write("0");
        bw.flush();
        bw.close();
    }
}